token-metadata/program/src/state.rs

```diff
@@ -81,6 +92,70 @@ pub struct Data {
     pub creators: Option<Vec<Creator>>,
 }
 
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+pub struct DataV2 {
+    /// The name of the asset
+    pub name: String,
+    /// The symbol for the asset
+    pub symbol: String,
+    /// URI pointing to JSON representing the asset
+    pub uri: String,
+    /// Royalty basis points that goes to creators in secondary sales (0-10000)
+    pub seller_fee_basis_points: u16,
+    /// Array of creators, optional
+    pub creators: Option<Vec<Creator>>,
+    /// Collection
+    pub collection: Option<Collection>,
+    /// Uses
+    pub uses: Option<Uses>,
+}
+
+impl DataV2 {
+    pub fn to_v1(&self) -> Data {
+        let ns = self.clone();
+        Data {
+            name: ns.name,
+            symbol: ns.symbol,
+            uri: ns.uri,
+            seller_fee_basis_points: ns.seller_fee_basis_points,
+            creators: ns.creators,
+        }
+    }
+}
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+pub enum UseMethod {
+    Burn,
+    Multiple,
+    Single,
+}
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+pub struct Uses {
+    pub use_method: UseMethod,
+    pub remaining: u64,
+    pub available: u64,
+}
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+pub enum TokenStandard {
+    NonFungible,  // This is a master edition
+    SemiFungible, // A token with metadata that can also have attrributes
+    Fungible,     // A token with simple metadata
+    Edition,      // This is a limited edition
+}
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+pub struct Collection {
```

Comment:
This should have `uri`, `name` and `num_item` fields as well.

`uri` for link to external JSON metadata scheme for Collection (image, description)

`name` for collection name

`num_items` for the number of items in the collection